scalafix-core/shared/src/main/scala/scalafix/internal/rule/DottyAutoTuplingFunctionArgs.scala

```diff
@@ -0,0 +1,53 @@
+package scalafix.internal.rule
+
+import scala.meta._
+import scala.meta.tokens.Token
+import scala.meta.tokens.Token._
+import scalafix.Patch
+import scalafix.rule.{Rule, RuleCtx}
+
+case object DottyAutoTuplingFunctionArgs
+    extends Rule("DottyAutoTuplingFunctionArgs") {
+  override def description: String =
+    "Rewrite that removes pattern-matching decomposition if function arguments can be automatically tupled in Dotty."
+
+  override def fix(ctx: RuleCtx): Patch =
+    ctx.tree.collect {
+      case Term.Apply((_, List(pf: Term.PartialFunction)))
+          if canBeAutoTupled(pf) =>
+        val pfTokens = pf.tokens
+
+        (for {
+          leftBrace <- pfTokens.headOption
```

Comment:
Guard against `.is[LeftBrace/RightBrace]`?

Reply:
I think the curly braces here are mandatory (http://www.lihaoyi.com/post/WartsoftheScalaProgrammingLanguage.html#needing-curliescase-for-destructuring-anonymous-functions). I put the guards anyway to make it explicit  👍